src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp

```diff
@@ -5,18 +5,57 @@
 #include "media.h"
 #include "marlin_vars.hpp"
 
-// M20 - List SD card
-void GcodeSuite::M20() {
-    SERIAL_ECHOLNPGM(MSG_BEGIN_FILE_LIST);
+static long GetFileSize(std::string filename) {
```

Comment:
This doesn't conform to coding style. Also, `std::string` allocates and is unacceptable here. Also, returning -1 or size seems wrong. Especially when we are printing the value.

Reply:
Done